### PR TITLE
Remove 'label' (eg frequent/hourly/daily etc) from snapshot name and place in snapshot property

### PR DESCRIPTION
**Pull request by leecallen**
Re: issue #10 "Integrate shadow copy snapshots for Samba"
Remove 'label' from snapshot name and place in a snapshot property.  

This change breaks the --fast option.  It may be possible to preserve that feature but it's beyond my scripting skills.

### CODE REVIEW

**Comment by maczkojanos** [comment]
Hi,

please explain: how to setup properly zfs-auto-snapshot for windows?
I'll try this : https://github.com/zfsonlinux/zfs-auto-snapshot/wiki/Samba , but i cant know, how to "fix a conflict".

Thanks
Janos

**Comment by stuehmer** [comment]
Hello Janos, "fixing the conflict" means looking at the file (after having completed the steps listed before) using a text editor and fixing the problem which is marked by the marker `<<<<<<<`. Those are conflicts that git could not fix automatically for example when two people edited the same line.

An explanation is e.g. here: http://genomewiki.ucsc.edu/index.php/Resolving_merge_conflicts_in_Git paragraph *How do I find conflicts within the file itself?* and the next paragraph.

File: src/zfs-auto-snapshot.sh [hunk 1]
```diff
@@ -186,7 +191,7 @@ do_snapshots () # properties, flags, snapname, oldglob, [targets...]
 					fi
 				fi
 			fi
-		done
+		done <<< "$SNAPSHOTS_OLD"
```
Comment (exenza) [inline src/zfs-auto-snapshot.sh]: I've merged @leecallen as per instructions on the wiki and I've changed to /bin/bash but now I'm getting echoed the error "The filesystem argument list is empty." (line 350 is the print)
I'm on Debian 8.8

[the remark this answers is not in the document]
Comment (rani-pinchuk) [reply src/zfs-auto-snapshot.sh]: Actually, when running zfs-auto-snapshot using bash without arguments, I did get the error that you mention ("The filesystem argument list is empty"). However, when running it with the correct arguments, it worked fine. I used the following arguments:
```
bash ./zfs-auto-snapshot --quiet --syslog --label=frequent --keep=4 //
```

**Comment by beren12** [comment]
Any chance this can get updated for latest master?
Or os there another tool that can do the same thing? I need static snapshot names for samba

**Comment by highvoltage** [comment]
Hi, thanks for the work that you've done on this before. If you're still interested, please submit a renewed patch against master since master has diverged too much already in the last few years. This upstream project has been dead for a while and I'm trying to revive it, so all input is appreciated.

**Comment by stuehmer** [comment]
@beren12 you won't need special snapshot names for Samba starting with Samba 4.6.2 which became more flexible. See https://github.com/zfsonlinux/zfs-auto-snapshot/wiki/Samba 'Update: Starting with Samba 4.6.2 [...]`

**Pull request closed without merging**